query/src/test/scala/filodb/query/exec/rangefn/AbsentFunctionSpec.scala

```diff
@@ -134,15 +138,20 @@ class AbsentFunctionSpec extends AnyFunSpec with Matchers with ScalaFutures with
     val columnFilter = Seq(ColumnFilter("host", Equals("host1")), ColumnFilter("instance", Equals("instance1")))
     val expectedKeys = Map(ZeroCopyUTF8String("host") -> ZeroCopyUTF8String("host1"),
       ZeroCopyUTF8String("instance") -> ZeroCopyUTF8String("instance1"))
-    val expectedRows = List((3000, 1.0))
+    val expectedRows = List((1000,Double.NaN), (2000,Double.NaN), (3000,1.0))
     val absentFunctionMapper = exec.AbsentFunctionMapper(columnFilter, RangeParams(1, 1, 3), "metric")
     val resultObs = absentFunctionMapper(Observable.fromIterable(testSampleNan), querySession, 1000, resultSchema, Nil)
     val result = resultObs.toListL.runToFuture.futureValue
     result.size shouldEqual (1)
     val keys = result.map(_.key.labelValues)
     val rows = result.flatMap(_.rows.map(x => (x.getLong(0), x.getDouble(1))).toList)
     keys.head shouldEqual expectedKeys
-    rows shouldEqual expectedRows
+    rows.zip(expectedRows).foreach {
+      case ((tsActual, vActual), (tsExp, vExp)) =>
+        tsActual shouldEqual tsExp
+        if (vExp.isNaN)    { vActual.isNaN shouldBe true }
+        else               { vActual shouldEqual vExp }
+    }
```

Comment:
Ignoring `zip`, isn't this the same as an `equals`? Foreach element:
1. If `exp=NaN`, make sure `act=NaN`
2. else make sure `exp=act`

In other words: "make sure each `exp=act`".

Could this diff be undone?

Reply:
Discussed offline -- explanation from @Boyuan-Yu:
> The logic is equivalent to a normal equal. Except for the NaN case where equal fails to compare them because Double.NaN == Double.NaN is false (which is Non-intuitive).